yarn-project/noir-protocol-circuits-types/src/scripts/demonomorphizer.ts

```diff
@@ -0,0 +1,274 @@
+import {
+  type AbiTypeWithGenerics,
+  type ArrayType,
+  BindingId,
+  type Constant,
+  type StringType,
+  type Struct,
+  type StructType,
+  type Tuple,
+  findAllStructsInType,
+  findStructsInType,
+} from './abi_type_with_generics.js';
+
+/**
+ * Demonomorphizes a list of ABI types adding generics to structs.
+ * Since monomorphization of the generics destroys information, this process is not guaranteed to return the original structure.
+ * However, it should succesfully unify all struct types that share the same name and field names.
+ */
+export class Demonomorphizer {
```

Comment:
Saw a lots of blue squiggles, apparently the spell checking plugin doesn't think this is a read word 😂  Might have to add all the variations to `cspell.json`.

Reply:
Will do!